File: 99e20e5e-3f2c-4cbd-90af-132279341356/Models/DTOs/Common/ApiResponse.cs
namespace FireIoTPlatform.Models.DTOs.Common;

public class ApiResponse<T>
{
    public int Code { get; set; } = 200;
    public string Message { get; set; } = "success";
    public T? Data { get; set; }
    public long Timestamp { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();

    public static ApiResponse<T> Success(T data)
    {
        return new ApiResponse<T> { Data = data };
    }

    public static ApiResponse<T> Success(string message, T data)
    {
        return new ApiResponse<T> { Message = message, Data = data };
    }

    public static ApiResponse<T> Error(int code, string message)
    {
        return new ApiResponse<T> { Code = code, Message = message };
    }
}

public class ApiResponse : ApiResponse<object>
{
    public static ApiResponse Success()
    {
        return new ApiResponse();
    }

    public new static ApiResponse Error(int code, string message)
    {
        return new ApiResponse { Code = code, Message = message };
    }
}

File: 787f5483-6f27-4954-a539-586982b42e25/backend/src/main/java/com/tvstation/media/common/ApiResponse.java
package com.tvstation.media.common;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ApiResponse<T> implements Serializable {

    private int code;
    private String message;
    private T data;
    private Long total;

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(200, "success", data, null);
    }

    public static <T> ApiResponse<T> success(T data, long total) {
        return new ApiResponse<>(200, "success", data, total);
    }

    public static <T> ApiResponse<T> success(String message, T data) {
        return new ApiResponse<>(200, message, data, null);
    }

    public static <T> ApiResponse<T> error(String message) {
        return new ApiResponse<>(500, message, null, null);
    }

    public static <T> ApiResponse<T> error(int code, String message) {
        return new ApiResponse<>(code, message, null, null);
    }

    public static <T> ApiResponse<T> success() {
        return new ApiResponse<>(200, "success", null, null);
    }
}
